core/src/epicli/data/common/ansible/playbooks/roles/preflight/tasks/main.yml

```diff
@@ -1,6 +1,32 @@
 ---
 - include_tasks: check-user.yml
 
+- name: Set is architecture supported to true
+  set_fact:
+     is_arch_supported: true
+  changed_when: true
+
+- name: Set is architecture supported to false if needed roles are not present in supported list of components
+  set_fact:
+     is_arch_supported: false
+  with_nested:
+    - "{{ supported_architecture }}"
+    - "{{ group_names }}"
+  when:
+    - item[1] not in item[0].components
+    - item[0].name == ansible_architecture
```

Comment:
It's possible to use intersection filter here without a loop
https://docs.ansible.com/ansible/latest/user_guide/playbooks_filters.html#selecting-from-sets-or-lists-set-theory

Reply:
used just difference